azurerm/internal/services/resource/subscription_template_deployment_resource.go

```diff
@@ -178,13 +178,11 @@ func subscriptionTemplateDeploymentResourceUpdate(d *schema.ResourceData, meta i
 		deployment.Properties.DebugSetting = expandTemplateDeploymentDebugSetting(d.Get("debug_level").(string))
 	}
 
-	if d.HasChange("parameters_content") {
-		parameters, err := expandTemplateDeploymentBody(d.Get("parameters_content").(string))
-		if err != nil {
-			return fmt.Errorf("expanding `parameters_content`: %+v", err)
-		}
-		deployment.Properties.Parameters = parameters
+	parameters, err := expandTemplateDeploymentBody(d.Get("parameters_content").(string))
+	if err != nil {
+		return fmt.Errorf("expanding `parameters_content`: %+v", err)
 	}
+	deployment.Properties.Parameters = parameters
```

Comment:
(same here) presumably this wants to use the default API value if it's ignored:

```suggestion
	if d.HasChange("parameters_content") {
		parameters, err := expandTemplateDeploymentBody(d.Get("parameters_content").(string))
		if err != nil {
			return fmt.Errorf("expanding `parameters_content`: %+v", err)
		}
		deployment.Properties.Parameters = parameters
	} else {
		deployment.Properties.Parameters = template.Properties.Parameters
	}
```